parboiled-core/src/main/scala/org/parboiled2/Parser.scala

```diff
@@ -466,15 +467,16 @@ abstract class Parser(initialValueStackSize: Int = 16,
   /**
    * THIS IS NOT PUBLIC API and might become hidden in future. Use only if you know what you are doing!
    */
-  def __matchMapWrapped(m: Map[String, Any]): Boolean = {
+  def __matchMapWrapped(m: Map[String, Any], ignoreCase: Boolean): Boolean = {
+    val matchString = if (ignoreCase) __matchIgnoreCaseStringWrapped _ else __matchStringWrapped _
```

Comment:
On Java7 (Scala 2.11) this allocates a new function instance _every_ time this method is executed, which can be quite costly.
Pulling the `if` into the `while` should be the better solution. If will be nicely predictable and thus only add negligible overhead.

Reply:
I fixed this one. Let me know when you're ready to merge. I'll squash commits beforehand.